middleware/proxy.go

```diff
@@ -54,6 +54,9 @@ type (
 		// Examples: If custom TLS certificates are required.
 		Transport http.RoundTripper
 
+		// ModifyRequest defines function to modify request before passing to ReverseProxy
```

Comment:
I think `Transport` field provides similar functionality as you can create your own `http.RoundTripper` which modifies request:

```go

type MyRoundTripper struct {
	accessToken string
	r           http.RoundTripper
}

func (mrt MyRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
	r.Header.Add("Authorization", "Bearer: "+mrt.accessToken)
	return mrt.r.RoundTrip(r)
}

func main() {
	e := echo.New()

	e.Use(middleware.ProxyWithConfig(middleware.ProxyConfig{
		Transport: MyRoundTripper{},
	}))

	log.Fatal(e.Start(":8080"))
}
```
https://stackoverflow.com/questions/51325704/adding-a-default-http-header-in-go